sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -1422,4 +1426,251 @@ object QueryExecutionErrors {
+         |structured streaming. While this was corrected in Spark 3.0, it appears that your
+         |query was started using an earlier version that incorrectly handled the checkpoint
+         |path.
+         |
+         |Correct Checkpoint Directory: $checkpointPath
+         |Incorrect Checkpoint Directory: $legacyCheckpointDir
+         |
+         |Please move the data from the incorrect directory to the correct one, delete the
+         |incorrect directory, and then restart this query. If you believe you are receiving
+         |this message in error, you can disable it with the SQL conf
+         |${SQLConf.STREAMING_CHECKPOINT_ESCAPED_PATH_CHECK_ENABLED.key}.
+       """.stripMargin)
+  }
+
+  def subProcessExitedError(
+      exitCode: Int, stderrBuffer: CircularBuffer, cause: Throwable): Throwable = {
+    new SparkException(s"Subprocess exited with status $exitCode. " +
+      s"Error: ${stderrBuffer.toString}", cause)
+  }
+
+  def outputDataTypeUnsupportedByNodeWithoutSerdeError(
+      nodeName: String, dt: DataType): Throwable = {
+    new SparkException(s"$nodeName without serde does not support " +
+      s"${dt.getClass.getSimpleName} as output data type")
+  }
+
+  def invalidStartIndexError(numRows: Int, startIndex: Int): Throwable = {
+    new ArrayIndexOutOfBoundsException(
+      "Invalid `startIndex` provided for generating iterator over the array. " +
+        s"Total elements: $numRows, requested `startIndex`: $startIndex")
+  }
+
+  def concurrentModificationOnExternalAppendOnlyUnsafeRowArrayError(
+      className: String): Throwable = {
+    new ConcurrentModificationException(
+      s"The backing $className has been modified since the creation of this Iterator")
+  }
+
+  def doExecuteBroadcastNotImplementedByNodeError(nodeName: String): Throwable = {
+    new UnsupportedOperationException(s"$nodeName does not implement doExecuteBroadcast")
+  }
+
+  def databaseNameConflictWithSystemPreservedDatabaseError(globalTempDB: String): Throwable = {
+    new SparkException(
+      s"""
+         |$globalTempDB is a system preserved database, please rename your existing database
+         |to resolve the name conflict, or set a different value for
+         |${GLOBAL_TEMP_DATABASE.key}, and launch your Spark application again.
+       """.stripMargin.split("\n").mkString(" "))
+  }
+
+  def commentOnTableUnsupportedError(): Throwable = {
+    new SQLFeatureNotSupportedException("comment on table is not supported")
+  }
+
+  def unsupportedUpdateColumnNullabilityError(): Throwable = {
+    new SQLFeatureNotSupportedException("UpdateColumnNullability is not supported")
+  }
+
+  def renameColumnUnsupportedForOlderMySQLError(): Throwable = {
+    new SQLFeatureNotSupportedException(
+      "Rename column is only supported for MySQL version 8.0 and above.")
+  }
+
+  def hitAnErrorWhenExecutingQueryError(e: Throwable): QueryExecutionException = {
```

Comment:
I feel this can be simplifed to `failedToExecuteQueryError`. Also for the return type: `QueryExecutionException` -> `Throwable`

Reply:
Please refer https://github.com/apache/spark/blob/960a7e5fceb2799882e5264540b2074ba9413e89/sql/core/src/main/scala/org/apache/spark/sql/util/QueryExecutionListener.scala#L64
So there need return `Exception`.